osu.Game/Rulesets/RulesetSelector.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Bindables;
+using osu.Framework.Allocation;
+
+namespace osu.Game.Rulesets
+{
+    public abstract class RulesetSelector : TabControl<RulesetInfo>
+    {
+        protected RulesetStore AvaliableRulesets;
+        protected readonly Bindable<RulesetInfo> GlobalRuleset = new Bindable<RulesetInfo>();
+
+        protected override Dropdown<RulesetInfo> CreateDropdown() => null;
+
+        /// <summary>
+        /// Whether we want to change a global ruleset when local one is changed.
+        /// </summary>
+        protected virtual bool AllowGlobalRulesetChange => true;
+
+        /// <summary>
+        /// Whether we want to change a local ruleset when global one is changed.
+        /// /// </summary>
+        protected virtual bool AllowLocalRulesetChange => true;
```

Comment:
Couldn't `Current` be used by its own for this? As in, rather than DI-ing in the parent ruleset, have the user set `Current = {ruleset bindable here}`, and always allow changing the ruleset unless it's disabled.

Reply:
In the future we want to make ruleset selector, which will be by it's own (https://github.com/ppy/osu/pull/4910). So in that case we don't want to change rulesets. That's why I don't use `Current` like that

Reply:
You still want the ruleset to change local to the profile overlay in that case though, right? You wouldn't pass in the global bindable there - at most you'd pass in a bindable local to the profile overlay.